### PR TITLE
[INLONG-7693][Sort] MySQL CDC Connector supports specifying field synchronization

### LINKED ISSUE #7693: [Improve][Sort] MySQL CDC Connector supports specifying field synchronization
### Description

When I use MySQL CDC Connector to synchronize specified fields for some tables, I expect to achieve the requirements by setting the `debezium.column.include.list` or `debezium.column.exclude.list` parameters. When creating the target table, only the fields I care about are used, and then the synchronization task execution fails. It only works when I create a target table that includes all the fields of the source table. Here I hope it can make the target table support only creating the fields I care about.

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by duowan1520**
### Prepare a Pull Request
- Fixes #7693 

### Motivation

MySQL CDC Connector supports specifying field synchronization

### CODE REVIEW

**Review by yunqingmoswu — APPROVED** [review]

**Comment by EMsnap** [comment]
Can you add a Test for this case?

**Comment by duowan1520** [comment]
> Can you add a Test for this case?
ok，i will do it

**Comment by duowan1520** [comment]
> @duowan1520 pls rebase from the master branch to fix the conflicts.

ok.

**Review by healchow — APPROVED** [review]

**Comment by EMsnap** [comment]
pls update the doc at the same time thx

**Comment by EMsnap** [comment]
BTW, we are planning to export the DDL to the sink connector to achieve the same action such as add a column;
What if the user filter out column A, and then alter column A with a DDL statement. Will the DDL on column A be filtered using the parameter u provide ?

**Comment by duowan1520** [comment]
> BTW, we are planning to export the DDL to the sink connector to achieve the same action such as add a column; What if the user filter out column A, and then alter column A with a DDL statement. Will the DDL on column A be filtered using the parameter u provide ?

The ddl statement seems to be generated before the getCanalData() method, I decided to wait until your pr is merged before proceeding

**Comment by duowan1520** [comment]
> BTW, we are planning to export the DDL to the sink connector to achieve the same action such as add a column; What if the user filter out column A, and then alter column A with a DDL statement. Will the DDL on column A be filtered using the parameter u provide ?

I will get back to you after understanding the PR related to export ddl

**Comment by duowan1520** [comment]
> BTW, we are planning to export the DDL to the sink connector to achieve the same action such as add a column; What if the user filter out column A, and then alter column A with a DDL statement. Will the DDL on column A be filtered using the parameter u provide ?

@EMsnap 
The current field filtering is implemented based on the comparison between the physical data column and TableChange. When the record is of ddl type, the current method cannot judge.

After I read the relevant export ddl pr, I have a supplementary idea, that is, `debezium.table.include.list`, `debezium.table.exclude.list`, `debezium.column.include.list`, `debezium .column.exclude.list` and other parameters are passed to MySqlRecordEmitter to determine whether the current ddl record is sent downstream.

What do you think of this implementation idea?

**Comment by duowan1520** [comment]
> BTW, we are planning to export the DDL to the sink connector to achieve the same action such as add a column; What if the user filter out column A, and then alter column A with a DDL statement. Will the DDL on column A be filtered using the parameter u provide ?

> 
Hi, the implementation has been adjusted this time. TableChange will remove the column that needs to be filtered before debeziumDeserializationSchema deserialization, and the downstream ddl statement can implement the corresponding filtering operation based on TableChange

**Comment by EMsnap** [comment]
@emhui PLAL at this pr , may be related to https://github.com/apache/inlong/pull/7750

File: inlong-sort/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/cdc/mysql/source/reader/MySqlRecordEmitter.java
Comment (duowan1520) [reply inlong-sort/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/cdc/mysql/source/reader/MySqlRecordEmitter.java:122]: There is a premise here. When the task sets the debezium.column.include.list/debezium.column.exclude.list parameter, it means that the task will only focus on those specified columns. When the filtered columns have dml and ddl changes , whose changes need to be ignored.

Therefore, when the columnFilter is applied here, the schema only retains the columns required by the task. In this way, the downstream can also filter out the columns that are not concerned in the ddl statement based on the new schema (todo)

**Review by e-mhui — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]
LGTM

**Pull request merged**
